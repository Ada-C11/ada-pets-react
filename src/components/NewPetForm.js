import React, { Component } from 'react';
import PropTypes from 'prop-types';

import 'bootstrap/dist/css/bootstrap.min.css';
import './NewPetForm.css'

class NewPetForm extends Component {
  constructor(props) {
    super(props);

    this.state = {
      name: "",
      species: "",
      location: "", 
      about: "", 
      images: ""
    };
  }
  
  addPet = (event) => {
    event.preventDefault();

    const pet = this.state;
    console.log(pet);

    this.props.addPetCallback(pet)
  }

  onInputChange = (event) => {
    const updatedState = {};
  
    const field = event.target.name;
    const value = event.target.value;
  
    updatedState[field] = value;
    this.setState(updatedState);
  }

  render() {
    return (
      <form  className="new-pet-form" onSubmit={this.addPet}>
        <h3>Add a Pet</h3>
        <label>
          Name: 
          <input 
            name="name" 
            type="text"
            value={this.state.name}
            onChange={this.onInputChange}></input>
        </label>
        <label>
          Species:
          <select 
            name="species"  
            value={this.state.species}
            onChange={this.onInputChange}>
              <option value="">Please Select a Species</option>
              <option value="cat">Cat</option>
              <option value="dog">Dog</option>
              <option value="other">Other</option>
          </select>
        </label>
        <label>
          Location: 
          <input 
            name="location"
            type="text"
            value={this.state.location}
            onChange={this.onInputChange}></input>
        </label>
        <label>
          About: 
          <textarea 
            name="about" 
            value={this.state.about}
            onChange={this.onInputChange} />
        </label>
        <label>
          Images: 
          <input name="images" type="text" value={this.state.images} onChange={this.onInputChange}></input>
        </label>
        <input className="btn btn-success new-pet-form--submit" type="submit" name="submit" value="Add a Pet" />
      </form>
    );
  }


}

NewPetForm.propTypes = {
  addPetCallback: PropTypes.func.isRequired,
};

export default NewPetForm;