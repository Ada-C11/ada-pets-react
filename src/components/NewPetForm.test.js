import React from 'react';
import { shallow } from 'enzyme'

import NewPetForm from './NewPetForm';


describe('New Pet Form', () => {
  test('it matches snapshot', () => {

    // Mount the component in the DOM
    const wrapper = shallow(<NewPetForm
      addPetCallback={() => { }}
    />);

    expect(wrapper).toMatchSnapshot();
  });
});