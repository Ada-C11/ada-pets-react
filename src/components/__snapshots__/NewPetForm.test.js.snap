// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`New Pet Form it matches snapshot 1`] = `
<form
  className="great-form"
  onSubmit={[Function]}
>
  <h3>
    Add a Pet
  </h3>
  <label>
    Name:
    <input
      name="name"
      onChange={[Function]}
      type="text"
      value=""
    />
  </label>
  <label>
    Species:
    <select
      name="species"
      onChange={[Function]}
      value=""
    >
      <option
        value=""
      >
        Please Select a Species
      </option>
      <option
        value="cat"
      >
        Cat
      </option>
      <option
        value="dog"
      >
        Dog
      </option>
      <option
        value="other"
      >
        Other
      </option>
    </select>
  </label>
  <label>
    Location:
    <input
      name="location"
      onChange={[Function]}
      type="text"
      value=""
    />
  </label>
  <label>
    About:
    <textarea
      name="about"
      onChange={[Function]}
      value=""
    />
  </label>
  <label>
    Images:
    <input
      name="images"
      onChange={[Function]}
      type="text"
      value=""
    />
  </label>
  <input
    className="btn btn-success new-pet-form--submit"
    name="submit"
    type="submit"
    value="Add a Pet"
  />
</form>
`;
